core/utils/eth_client_utils.go

```diff
@@ -42,3 +48,46 @@ func BytesToQuorumThresholdPercentages(quorumThresholdPercentagesBytes []byte) e
 	}
 	return quorumThresholdPercentages
 }
+
+// Very basic algorithm to calculate the gasPrice bump based on the currentGasPrice and retry iteration.
+// It adds a i/10 percentage to the current prices, where i represents the iteration.
+func CalculateGasPriceBumpBasedOnRetry(currentGasPrice *big.Int, iteration int) *big.Int {
+	factor := (new(big.Int).Add(big.NewInt(100), new(big.Int).Mul(big.NewInt(int64(iteration)), big.NewInt(10))))
+	gasPrice := new(big.Int).Mul(currentGasPrice, factor)
+	gasPrice = gasPrice.Div(gasPrice, big.NewInt(100))
```

Comment:
very small nit :), but it could make the math a little clearer. We can make two variables with names like `factorNumerator` and `factorDenominator`, and then the gas price will be
```
gasPrice := currentGasPrice * factorNumerator / factorDenominator
```

Reply:
Updated! Though `big.Int` math is horrible to work with, you can't use the normal operators.